samcli/commands/deploy/guided_context.py

```diff
@@ -285,52 +288,226 @@ def prompt_parameters(
+            manager_helper.missing_repo_functions
+        )
+        click.echo(
+            "\t Image repositories: "
+            f"Found ({functions_with_repo_count} of {len(manager_helper.function_logical_ids)})"
+            " #Different image repositories can be set in samconfig.toml"
+        )
+
+        if not manager_helper.missing_repo_functions:
+            return False
+
+        click.echo(
+            "\t #Managed repositories will be deleted when their functions are "
+            "removed from the template and deployed"
+        )
+        return (
+            confirm(
+                f"\t {self.start_bold}Create managed ECR repositories for the "
+                f"{len(manager_helper.missing_repo_functions)} functions without?{self.end_bold}",
+                default=True,
+            )
+            if manager_helper.missing_repo_functions
+            else True
+        )
+
+    def prompt_delete_unreferenced_repos(
+        self, manager_helper: CompanionStackManagerHelper, image_repositories: Dict[str, str]
+    ) -> Dict[str, str]:
+        """
+        Prompt user for deleting unreferenced companion stack image repos.
+        Throws GuidedDeployFailedError if delete repos has been denied by the user.
+        This function does not actually remove the functions from the stack.
+
+        Parameters
+        ----------
+        manager_helper: CompanionStackManagerHelper
+            Instance of CompanionStackManagerHelper
+
+        image_repositories: Dict[str, str]
+            Current image repo dictionary with function logical ID as key and image repo URI as value.
+
+        Returns
+        -------
+        Dict[str, str]
+            Updated image repo dictionary with unreferenced repos removed
+        """
+        if not manager_helper.unreferenced_repos:
+            return image_repositories
+
+        click.echo(
+            "\t Checking for unreferenced ECR repositories to clean-up: "
+            f"{len(manager_helper.unreferenced_repos)} found"
+        )
+        for repo in manager_helper.unreferenced_repos:
+            repo_uri = manager_helper.manager.get_repo_uri(repo)
+            click.echo(f"\t  {repo_uri}")
+        delete_repos = confirm(
+            f"\t {self.start_bold}Delete the unreferenced repositories listed above when deploying?{self.end_bold}",
+            default=False,
+        )
+        if not delete_repos:
+            click.echo("\t Deployment aborted!")
+            click.echo(
+                "\t #The deployment was aborted to prevent "
+                "unreferenced managed ECR repositories from being deleted.\n"
+                "\t #You may remove repositories from the SAMCLI "
+                "managed stack to retain them and resolve this unreferenced check."
+            )
+            raise GuidedDeployFailedError("Unreferenced Auto Created ECR Repos Must Be Deleted.")
+
+        return manager_helper.remove_unreferenced_repos_from_mapping(image_repositories)
```

Comment:
the prompts and the actions taken from the prompts seem to be tightly coupled here, maybe we could break them apart.

Reply:
Updated prompt methods (prompt_specify_repos, prompt_create_all_repos, and prompt_delete_unreferenced_repos) to be less coupled.